diff --git a/Benchmarking.org b/Benchmarking.org
index 80a9cc4..650dfe3 100644
--- a/Benchmarking.org
+++ b/Benchmarking.org
@@ -63,6 +63,7 @@ file, and produces commands to run the benchmarks.
     import datetime
     import subprocess
     import os
+    import re
     from utils import run_commands, get_config_summary
 
     <<parsing-dacapo-output>>
diff --git a/Makefile b/Makefile
new file mode 100644
index 0000000..4fe9e1a
--- /dev/null
+++ b/Makefile
@@ -0,0 +1,42 @@
+export PYTHONPATH=.
+PYTHON=python3
+
+SCRIPTS=variants/scripts
+
+# chart
+variants/naive-dacapo-2006-chart.jar: programs/dacapo-2006-10-MR2-noexit.jar
+	${PYTHON} ${SCRIPTS}/chart-naive.py
+
+variants/identity-dacapo-chart.jar: programs/dacapo-2006-10-MR2-noexit.jar
+	${PYTHON} ${SCRIPTS}/chart-identity.py
+
+variants/coco-dacapo-chart.jar: programs/dacapo-2006-10-MR2-noexit.jar
+	${PYTHON} ${SCRIPTS}/chart-coco.py
+
+variants/cswitch-dacapo-chart.jar: programs/dacapo-2006-10-MR2-noexit.jar
+	${PYTHON} ${SCRIPTS}/chart-cswitch.py
+
+variants/jbrainy-dacapo-2006-chart.jar: programs/dacapo-2006-10-MR2-noexit.jar
+	${PYTHON} ${SCRIPTS}/chart-jbrainy.py
+
+chart-naive: variants/naive-dacapo-2006-chart.jar
+chart-identity: variants/identity-dacapo-chart.jar
+chart-coco: variants/coco-dacapo-chart.jar
+chart-cswitch: variants/cswitch-dacapo-chart.jar
+chart-jbrainy: variants/jbrainy-dacapo-2006-chart.jar
+
+# fop
+variants/identity-dacapo-fop.jar: programs/dacapo-fop.jar
+	${PYTHON} ${SCRIPTS}/fop-identity.py
+
+variants/naive-dacapo-fop.jar: programs/dacapo-fop.jar
+	${PYTHON} ${SCRIPTS}/fop-naive.py
+
+variants/jbrainy-dacapo-fop.jar: programs/dacapo-fop.jar
+	${PYTHON} ${SCRIPTS}/fop-jbrainy.py
+
+fop-identity: variants/identity-dacapo-fop.jar
+fop-naive: variants/naive-dacapo-fop.jar
+fop-jbrainy: variants/jbrainy-dacapo-fop.jar
+
+all: chart-naive chart-identity chart-coco chart-cswitch chart-jbrainy fop-identity fop-naive fop-jbrainy
diff --git a/benchmarking.py b/benchmarking.py
index b393648..9fa7259 100644
--- a/benchmarking.py
+++ b/benchmarking.py
@@ -2,6 +2,7 @@ import json
 import datetime
 import subprocess
 import os
+import re
 from utils import run_commands, get_config_summary
 
 def classify_line(line):
diff --git a/variants/scripts/chart-coco.py b/variants/scripts/chart-coco.py
new file mode 100644
index 0000000..49f702f
--- /dev/null
+++ b/variants/scripts/chart-coco.py
@@ -0,0 +1,6 @@
+import main
+
+print("Building chart-coco")
+main.build_coco_dacapo_variant("programs/dacapo-2006-10-MR2-noexit.jar",
+                               "coco-dacapo-chart.jar",
+                               benchmark_dir="org/jfree")
diff --git a/variants/scripts/chart-cswitch.py b/variants/scripts/chart-cswitch.py
new file mode 100644
index 0000000..8661f78
--- /dev/null
+++ b/variants/scripts/chart-cswitch.py
@@ -0,0 +1,6 @@
+import main
+
+print("Building chart-cswitch")
+main.build_cswitch_variant("programs/dacapo-2006-10-MR2-noexit.jar",
+                           "cswitch-dacapo-chart.jar",
+                           benchmark_dir="org/jfree")
diff --git a/variants/scripts/chart-identity.py b/variants/scripts/chart-identity.py
new file mode 100644
index 0000000..81fd4fd
--- /dev/null
+++ b/variants/scripts/chart-identity.py
@@ -0,0 +1,10 @@
+from identity import replace_classes_identity
+from utils import run_commands
+from dacapo import create_variant
+
+print("Building chart-identity")
+create_variant("programs/dacapo-2006-10-MR2-noexit.jar",
+               replace_classes_identity,
+               "variants/identity-dacapo-chart.jar",
+               adapt_harness=replace_classes_identity,
+               benchmark_dir="org/jfree")
\ No newline at end of file
diff --git a/variants/scripts/chart-jbrainy.py b/variants/scripts/chart-jbrainy.py
new file mode 100644
index 0000000..4f1d3ca
--- /dev/null
+++ b/variants/scripts/chart-jbrainy.py
@@ -0,0 +1,8 @@
+from main import build_jbrainy_dacapo_variant
+
+print("Building chart-jbrainy")
+build_jbrainy_dacapo_variant("programs/dacapo-2006-10-MR2-noexit.jar",
+                             "chart",
+                             "jbrainy-dacapo-2006-chart.jar",
+                             "PAPI",
+                             benchmark_dir="org/jfree")
diff --git a/variants/scripts/chart-naive.py b/variants/scripts/chart-naive.py
new file mode 100644
index 0000000..9564aa5
--- /dev/null
+++ b/variants/scripts/chart-naive.py
@@ -0,0 +1,9 @@
+from main import build_jbrainy_dacapo_variant
+
+print("Building chart-naive")
+build_jbrainy_dacapo_variant("programs/dacapo-2006-10-MR2-noexit.jar",
+                             "chart",
+                             "naive-dacapo-2006-chart.jar",
+                             "BENCHMARK",
+                             benchmark_dir="org/jfree")
+
diff --git a/variants/scripts/fop-identity.py b/variants/scripts/fop-identity.py
new file mode 100644
index 0000000..5998b48
--- /dev/null
+++ b/variants/scripts/fop-identity.py
@@ -0,0 +1,7 @@
+from identity import replace_classes_identity
+from utils import run_commands
+from dacapo import create_variant
+
+create_variant("programs/dacapo-fop.jar",
+               replace_classes_identity,
+               "variants/identity-dacapo-fop.jar")
\ No newline at end of file
diff --git a/variants/scripts/fop-jbrainy.py b/variants/scripts/fop-jbrainy.py
new file mode 100644
index 0000000..f8202e2
--- /dev/null
+++ b/variants/scripts/fop-jbrainy.py
@@ -0,0 +1,7 @@
+from main import build_jbrainy_dacapo_variant
+
+print("Building fop-jbrainy")
+build_jbrainy_dacapo_variant("programs/dacapo-fop.jar",
+                             "fop",
+                             "jbrainy-dacapo-fop.jar",
+                             "PAPI")
diff --git a/variants/scripts/fop-naive.py b/variants/scripts/fop-naive.py
new file mode 100644
index 0000000..2f6da32
--- /dev/null
+++ b/variants/scripts/fop-naive.py
@@ -0,0 +1,7 @@
+from main import build_jbrainy_dacapo_variant
+
+print("Building naive-fop")
+build_jbrainy_dacapo_variant("programs/dacapo-fop.jar",
+                             "fop",
+                             "naive-dacapo-fop.jar",
+                             "BENCHMARK")
