diff --git a/spotbugs-ant/src/main/java/edu/umd/cs/findbugs/anttask/FindBugsTask.java b/spotbugs-ant/src/main/java/edu/umd/cs/findbugs/anttask/FindBugsTask.java
index 3a8d79379..c2e24430f 100644
--- a/spotbugs-ant/src/main/java/edu/umd/cs/findbugs/anttask/FindBugsTask.java
+++ b/spotbugs-ant/src/main/java/edu/umd/cs/findbugs/anttask/FindBugsTask.java
@@ -761,8 +761,9 @@ public class FindBugsTask extends AbstractFindBugsTask {
         if (outputFormat != null
                 && !("xml".equalsIgnoreCase(outputFormat.trim()) || "xml:withMessages".equalsIgnoreCase(outputFormat.trim())
                         || "html".equalsIgnoreCase(outputFormat.trim()) || "text".equalsIgnoreCase(outputFormat.trim())
+                        || "sarif".equalsIgnoreCase(outputFormat.trim())
                         || "xdocs".equalsIgnoreCase(outputFormat.trim()) || "emacs".equalsIgnoreCase(outputFormat.trim()))) {
-            throw new BuildException("output attribute must be either " + "'text', 'xml', 'html', 'xdocs' or 'emacs' for task <"
+            throw new BuildException("output attribute must be either " + "'text', 'xml', 'html', 'xdocs', 'sarif' or 'emacs' for task <"
                     + getTaskName() + "/>", getLocation());
         }
 
